core/codegen/src/derive/from_param.rs

```diff
@@ -0,0 +1,42 @@
+use devise::Support;
+use proc_macro2::TokenStream;
+use quote::quote;
+use devise::*;
+use devise::ext::SpanDiagnosticExt;
+use crate::exports::*;
+
+pub fn derive_from_param(input: proc_macro::TokenStream) -> TokenStream {
+    DeriveGenerator::build_for(input, quote!(impl<'a> #_request::FromParam<'a>))
+        .support(Support::Enum)
+        .validator(ValidatorBuild::new()
+                   .fields_validate(|_, fields| {
+                       if !fields.is_empty() {
+                           return Err(fields.span().error("Only empty enums are accepted"));
+                       }
+                       Ok(())
+                   })
+        )
+        .inner_mapper(MapperBuild::new()
+                      .enum_map(|_, data| {
+                          let mut matches = vec![];
+
+                          for field in data.variants() {
+                              let field_name = &field;
+                              matches.push(quote!(
+                                  stringify!(#field_name) => Ok(Self::#field_name),
+                              ))
+                          }
```

Comment:
This can probably be a `.iter().map()`. I don't recall if we quote can use the iterator directly, but if it can, then that would save us the allocation here.